### ISSUE TITLE
Github "issues commented" includes comments by other users

### ISSUE THREAD

**Issue by nikic** [psss/did#325]
The "issues commented" are currently computed like this: https://github.com/psss/did/blob/e1e3de7a7c7c3d80484292c9e926651d3baafdd1/did/plugins/github.py#L168

What this gives you is any issues that a) I commented on and b) that were updated in the date period. This means that if I commented on an issue at some point, and then someone else leaves a comment on it in the date period, then it will be reported as if I commented that week, which is probably not what was intended?

The same issue exists for "pull requests commented".

**Comment by nikic** [comment]
I think there are other related issues as well, e.g. it looks like "issues closed" is determined based on issues closed in the time frame where I am the assignee, rather than issues that I closed.

It's probably not possible to implemented correct behavior using the search API. I found that there is an API for getting events for a user: https://docs.github.com/en/rest/activity/events?apiVersion=2022-11-28#list-public-events-for-a-user This does seem to provide the necessary information to correctly determine all the issues/PRs a user interacted with in a specific way.

However, that API does not seem to allow specifying a time range, so one would have to scan the feed backwards from the current time. This would only work for recent date ranges.

**Comment by psss** [comment]
Yes, sometimes the stats are tricky and hard to be extracted correctly. For some plugins we do some additional filtering of the detailed history. We could perhaps do something similar for the `github` plugin as well.

**Comment by Felixoid** [comment]
Hi-hi! I started using the program recently and love it!

However, I faced precisely the same issue described here. Reading the GH API documentation doesn't help. The events API does not provide a way to filter them by dates.

What we can use is additional filtering for the comments and review comments like here:

https://api.github.com/repos/ClickHouse/ClickHouse/issues/56317/comments?since=2023-11-15&sort=created
https://api.github.com/repos/ClickHouse/ClickHouse/pulls/56317/comments?since=2023-11-15&sort=created

The first link is for normal comments in the PRs and issues, and the latter one is for the review comments.

**Comment by Felixoid** [comment]
I've implemented the filter in #363, but I've found that the `PullRequestsReviewed` has a very solid query, so it's unnecessary to use https://api.github.com/repos/ClickHouse/ClickHouse/pulls/56317/comments?since=2023-11-15&sort=created there

**Comment by pcahyna** [comment]
> I've implemented the filter in [#363](https://github.com/psss/did/pull/363), but I've found that the `PullRequestsReviewed` has a very solid query, so it's unnecessary to use https://api.github.com/repos/ClickHouse/ClickHouse/pulls/56317/comments?since=2023-11-15&sort=created there

I think you need it, because the query is very restrictive and one misses review comments. See #465
